source/fundamentals/monitoring.txt

```diff
@@ -1,7 +1,435 @@
+           "HeartbeatInterval": 0,
+           "HelloOK": false,
+           "Hosts": null,
+           "LastError": null,
+           "LastUpdateTime": "...",
+           "LastWriteTime": "...",
+           "MaxBatchCount": 0,
+           "MaxDocumentSize": 0,
+           "MaxMessageSize": 0,
+           "Members": null,
+           "Passives": null,
+           "Passive": false,
+           "Primary": "...",
+           "ReadOnly": false,
+           "ServiceID": null,
+           "SessionTimeoutMinutes": 0,
+           "SetName": "...",
+           "SetVersion": 0,
+           "Tags": null,
+           "TopologyVersion": null,
+           "Kind": 0,
+           "WireVersion": null
+       },
+       "NewDescription": {
+           "Addr": "...",
+           "Arbiters": null,
+           "AverageRTT": ...,
+           "AverageRTTSet": true,
+           "Compression": null,
+           "CanonicalAddr": "...",
+           "ElectionID": "...",
+           "HeartbeatInterval": ...,
+           "HelloOK": true,
+           "Hosts": [...],
+           "LastError": null,
+           "LastUpdateTime": "...",
+           "LastWriteTime": "...",
+           "MaxBatchCount": ...,
+           "MaxDocumentSize": ...,
+           "MaxMessageSize": ...,
+           "Members": [...],
+           "Passives": null,
+           "Passive": false,
+           "Primary": "...",
+           "ReadOnly": false,
+           "ServiceID": null,
+           "SessionTimeoutMinutes": 30,
+           "SetName": "...",
+           "SetVersion": 9,
+           "Tags": [...],
+           "TopologyVersion": {...},
+           "Kind": 10,
+           "WireVersion": {...}
+       }
+   }
+
+The ``Kind`` field in the server descriptions represents the type of a
+single server in a topology and can have the following values:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Value
+     - Description
+   * - ``0``
+     - Unknown instance
+   * - ``1``
+     - Standalone instance
+   * - ``2``
+     - Replica set member
+   * - ``6``
+     - Primary instance
+   * - ``10``
+     - Secondary instance
+   * - ``18``
+     - Arbiter instance
+   * - ``34``
+     - Replica set ghost, a member that cannot be queried
+   * - ``256``
+     - Mongos proxy instance
+   * - ``512``
+     - Load balancer instance
+
+ServerOpeningEvent
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+   :copyable: false
+
+   *event.ServerOpeningEvent
+   {
+       "Address": "...",
+       "TopologyID": "..."
+   }
+
+ServerClosedEvent
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: bash
+   :copyable: false
+
+   *event.ServerClosedEvent
+   {
+       "Address": "...",
+       "TopologyID": "..."
+   }
+
+TopologyDescriptionChangedEvent
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. important::
+
+   The driver calls ``TopologyDescriptionChanged`` when the deployment topology is locked,
+   so the callback should not attempt any operation that requires server
+   selection on the same client.
```

Comment:
```suggestion
   Because the driver calls ``TopologyDescriptionChanged`` when the deployment topology is locked,
   the callback for this event should not attempt any operation that requires server
   selection on the same client.
```